[pcre] build failure

**Host Environment**
Win10 x64/VS2019 (latest version as of 5.11.2021)
vcpkg git

**To Reproduce**
Steps to reproduce the behavior:
```
git clone https://github.com/microsoft/vcpkg.git
cd vcpkg
.\vcpkg update
.\vcpkg integrate install
.\vcpkg install --triplet x64-windows pcre

```
vcpkg tries to use the old pcre download-server: https://ftp.pcre.org/pub/pcre/pcre-8.45.zip

i first thought it was language pack related, but it even don't work with english language pack installed

**Failure logs** 
```
D:\temp\dosbox_dev\vcpkg>.\vcpkg install --triplet x64-windows pcre
Computing installation plan...
The following packages will be built and installed:
    pcre[core]:x64-windows -> 8.45
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\Dennis\AppData\Local\vcpkg\archives in 100.5 us. Use --debug to see more details.
Starting package 1/1: pcre:x64-windows
Building package pcre[core]:x64-windows...
CMake Warning at scripts/cmake/vcpkg_download_distfile.cmake:169 (message):
  SILENT_EXIT has been deprecated as an argument to vcpkg_download_distfile
  -- remove the argument to resolve this warning
Call Stack (most recent call first):
  ports/pcre/portfile.cmake:12 (vcpkg_download_distfile)
  scripts/ports.cmake:142 (include)


-- Downloading https://ftp.pcre.org/pub/pcre/pcre-8.45.zip -> pcre-8.45.zip...
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://ftp.pcre.org/pub/pcre/pcre-8.45.zip
[DEBUG] Download failed -- retrying after 1000 ms.
[DEBUG] Download failed -- retrying after 2000 ms.
[DEBUG] Download failed -- retrying after 4000 ms.
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(656)
Error: Failed to download from mirror set:
https://ftp.pcre.org/pub/pcre/pcre-8.45.zip: WinHttpSendRequest() failed: 12002
https://ftp.pcre.org/pub/pcre/pcre-8.45.zip: WinHttpSendRequest() failed: 12002
https://ftp.pcre.org/pub/pcre/pcre-8.45.zip: WinHttpSendRequest() failed: 12002
https://ftp.pcre.org/pub/pcre/pcre-8.45.zip: WinHttpSendRequest() failed: 12002

[DEBUG] Exiting after %s (%d us)


CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:83 (message):

      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.

      3. You proxy's remote server is out of service.

      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)

      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:291 (z_vcpkg_download_distfile_show_proxy_and_fail)
  ports/pcre/portfile.cmake:12 (vcpkg_download_distfile)
  scripts/ports.cmake:142 (include)


Error: Building package pcre:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: pcre[core]:x64-windows -> 8.45
  vcpkg version: 2021-11-02-af04ebf6274fd6f7a941bff4662b3955c64f6f42
  vcpkg-tool version: 8acb35e27 2021-11-05 (58 minutes ago)

Additionally, attach any relevant sections from the log files above.

D:\temp\dosbox_dev\vcpkg>
```

so the server is not reachable as state on pcre.org

> Note that the former ftp.pcre.org FTP site is no longer available. You will need to update any scripts that download PCRE source code to download via HTTPS, Git, or Subversion from the new home on GitHub instead.

Using #21193, I am experiencing the following error with pcre2 file hash mismatch:

```
Starting package 1/4: pcre2:x64-windows
Building package pcre2[core]:x64-windows...
-- Downloading https://github.com/PhilipHazel/pcre2/archive/pcre2-10.37.tar.gz -> PhilipHazel-pcre2-pcre2-10.37.tar.gz...
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://github.com/PhilipHazel/pcre2/archive/pcre2-10.37.tar.gz
Error: Failed to download from mirror set:
File does not have the expected hash:
             url : [ https://github.com/PhilipHazel/pcre2/archive/pcre2-10.37.tar.gz ]
       File path : [ C:\dev\vcpkg\vcpkg\downloads\PhilipHazel-pcre2-pcre2-10.37.tar.gz.2328.part ]
   Expected hash : [ f91760a8e0747f52211612fb0e134d685e224d16bd884eb574718d077a586b1fd7b6435d4e3b75c879b12e02b252467ecc28cdc4bc2903c783dacab089f99c99 ]
     Actual hash : [ f15357d8f5400739f26a32fe03a0759607fd7a282a1a8a2bb169b971a8a00330eff3ba465842c4668ad968e3fd1fff4dc7f0631d0af8416947070ff7cd056e6f ]


[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(626)
[DEBUG] Exiting after 2.952 s us (2946634 us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:84 (message):

      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.

      3. You proxy's remote server is out of service.

      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)

      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:292 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:173 (vcpkg_download_distfile)
  ports/pcre2/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:142 (include)


Error: Building package pcre2:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: pcre2:x64-windows
  Vcpkg version: 2021-09-10-2059ef11aa6067e6f59b0d939c5d17e3c5c47d3e
```

--- a/ports/pcre/portfile.cmake
+++ b/ports/pcre/portfile.cmake
@@ -9,33 +9,17 @@ set(PATCHES
         pcre-8.45_fix_postfix_for_debug_Windows_builds.patch
         export-cmake-targets.patch)
 
-vcpkg_download_distfile(ARCHIVE
-    URLS "https://ftp.pcre.org/pub/pcre/pcre-${PCRE_VERSION}.zip"
+vcpkg_from_sourceforge(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO pcre/pcre
+    REF ${PCRE_VERSION}
     FILENAME "pcre-${PCRE_VERSION}.zip"
     SHA512 ${EXPECTED_SHA}
-    SILENT_EXIT
+    PATCHES ${PATCHES}
 )
 
-if (EXISTS "${ARCHIVE}")
-    vcpkg_extract_source_archive_ex(
-        OUT_SOURCE_PATH SOURCE_PATH
-        ARCHIVE ${ARCHIVE}
-        PATCHES ${PATCHES}
-    )
-else()
-    vcpkg_from_sourceforge(
-        OUT_SOURCE_PATH SOURCE_PATH
-        REPO pcre/pcre
-        REF ${PCRE_VERSION}
-        FILENAME "pcre-${PCRE_VERSION}.zip"
-        SHA512 ${EXPECTED_SHA}
-        PATCHES ${PATCHES}
-    )
-endif()
-
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
     OPTIONS
         -DPCRE_BUILD_TESTS=NO
         -DPCRE_BUILD_PCREGREP=NO
@@ -55,11 +39,11 @@ vcpkg_configure_cmake(
     # OPTIONS_DEBUG -DDEBUGGABLE=1
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/unofficial-${PORT} TARGET_PATH share/unofficial-${PORT})
+vcpkg_cmake_config_fixup(PACKAGE_NAME "unofficial-${PORT}" CONFIG_PATH "share/unofficial-${PORT}")
 
-foreach(FILE ${CURRENT_PACKAGES_DIR}/include/pcre.h ${CURRENT_PACKAGES_DIR}/include/pcreposix.h)
+foreach(FILE "${CURRENT_PACKAGES_DIR}/include/pcre.h" "${CURRENT_PACKAGES_DIR}/include/pcreposix.h")
     file(READ ${FILE} PCRE_H)
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
         string(REPLACE "defined(PCRE_STATIC)" "1" PCRE_H "${PCRE_H}")
@@ -71,18 +55,17 @@ endforeach()
 
 vcpkg_fixup_pkgconfig()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/man)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/man)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/share/doc)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/man")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/man")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/doc")
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
 endif()
 
 vcpkg_copy_pdbs()
-configure_file(${CMAKE_CURRENT_LIST_DIR}/unofficial-pcre-config.cmake ${CURRENT_PACKAGES_DIR}/share/unofficial-pcre/unofficial-pcre-config.cmake @ONLY)
+configure_file("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcre-config.cmake" "${CURRENT_PACKAGES_DIR}/share/unofficial-pcre/unofficial-pcre-config.cmake" @ONLY)
 
-file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/pcre/vcpkg.json
+++ b/ports/pcre/vcpkg.json
@@ -1,6 +1,17 @@
 {
   "name": "pcre",
   "version": "8.45",
+  "port-version": 1,
   "description": "Perl Compatible Regular Expressions",
-  "homepage": "https://www.pcre.org/"
+  "homepage": "https://www.pcre.org/",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ]
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5122,7 +5122,7 @@
     },
     "pcre": {
       "baseline": "8.45",
-      "port-version": 0
+      "port-version": 1
     },
     "pcre2": {
       "baseline": "10.37",

--- a/versions/p-/pcre.json
+++ b/versions/p-/pcre.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b1eb815e0c923baec32da26918e951b288adbaa5",
+      "version": "8.45",
+      "port-version": 1
+    },
+    {
       "git-tree": "665bab8c2bf878815782cff93dfff5720316ab88",
       "version": "8.45",
       "port-version": 0

